Documentation: BuildToolPlugin example uses wrong type for executable

With swift from swiftly main-snapshot-2025-06-03, BuildCommand expects `executable` to be a URL.

https://github.com/swiftlang/swift-package-manager/blob/ff7ea75b95e503bd3f22fcf8cf09e274561c3c10/Sources/PackagePlugin/Command.swift#L45-L53

Swift version 6.2-dev (LLVM b5d039be1fbae13, Swift 4fb4945ab972c85)
Target: x86_64-unknown-linux-gnu
Build config: +assertions

The example in Plugins.md passes a `Path` instead.

https://github.com/swiftlang/swift-package-manager/blob/ff7ea75b95e503bd3f22fcf8cf09e274561c3c10/Documentation/Plugins.md#implementing-the-build-tool-plugin-script

```swift
return .buildCommand(
    displayName: "Generating \(outputName) from \(inputPath.lastComponent)",
=>  executable: try context.tool(named: "SomeTool").path,
    arguments: [ "--verbose", "\(inputPath)", "\(outputPath)" ],
    inputFiles: [ inputPath, ],
    outputFiles: [ outputPath ]
)
```

Trying this pattern in my own code is a hard error.

```
Swift version 6.2-dev (LLVM b5d039be1fbae13, Swift 4fb4945ab972c85)
Target: x86_64-unknown-linux-gnu
/home/andrew/ladybird-org/htmlparser-swift/Plugins/GenerateNamedCharacterReferencesPlugin/main.swift:35:89: error: cannot convert value of type 'Path' to expected argument type 'URL'
33 |             .buildCommand(
34 |                 displayName: "Generate Named Character References",
35 |                 executable: try context.tool(named: "GenerateNamedCharacterReferences").path,
   |                                                                                         `- error: cannot convert value of type 'Path' to expected argument type 'URL'
36 |                 arguments: [
37 |                     "--generated-header-path", outputHeader.path,
error: build planning stopped due to build-tool plugin failures
```

--- a/Sources/PackageManagerDocs/Documentation.docc/Plugins/WritingBuildToolPlugin.md
+++ b/Sources/PackageManagerDocs/Documentation.docc/Plugins/WritingBuildToolPlugin.md
@@ -123,23 +123,36 @@ struct MyPlugin: BuildToolPlugin {
     
   func createBuildCommands(context: PluginContext, 
                            target: Target) throws -> [Command] {
-    guard let target = target.sourceModule else { return [] }
-    let inputFiles = target.sourceFiles.filter(
-      { $0.path.extension == "dat" }
-    )
-    return try inputFiles.map {
-      let inputFile = $0
-      let inputPath = inputFile.path
-      let outputName = inputPath.stem + ".swift"
-      let outputPath = context.pluginWorkDirectory.appending(outputName)
-      return .buildCommand(
-          displayName: "Generating \(outputName) from \(inputPath.lastComponent)",
-          executable: try context.tool(named: "SomeTool").path,
-          arguments: [ "--verbose", "\(inputPath)", "\(outputPath)" ],
-          inputFiles: [ inputPath, ],
-          outputFiles: [ outputPath ]
-      )
+    // This plugin only runs for package targets that can have source files.
+    guard let sourceFiles = target.sourceModule?.sourceFiles else { return [] }
+
+    // Find the code generator tool to run (replace this with the actual one).
+    let generatorTool = try context.tool(named: "my-code-generator")
+
+    // Construct a build command for each source file with a particular suffix.
+    return sourceFiles.map(\.url).compactMap {
+        createBuildCommand(for: $0, in: context.pluginWorkDirectoryURL, with: generatorTool.url)
     }
+  }
+
+  func createBuildCommand(for inputPath: URL,
+                          in outputDirectoryPath: URL,
+                          with generatorToolPath: URL) -> Command? {
+    // Skip any file that doesn't have the extension we're looking for
+    // (replace this with the actual one).
+    guard inputPath.pathExtension == "my-input-suffix" else { return .none }
+
+    // Return a command that will run during the build to generate the output file.
+    let inputName = inputPath.lastPathComponent
+    let outputName = inputPath.deletingPathExtension().lastPathComponent + ".swift"
+    let outputPath = outputDirectoryPath.appendingPathComponent(outputName)
+    return .buildCommand(
+        displayName: "Generating \(outputName) from \(inputName)",
+        executable: generatorToolPath,
+        arguments: ["\(inputPath)", "-o", "\(outputPath)"],
+        inputFiles: [inputPath],
+        outputFiles: [outputPath]
+    )
   }
 }
 ```
@@ -163,9 +176,11 @@ struct MyBuildToolPlugin: BuildToolPlugin {
     // This example configures `sometool` to write to a 
     // "GeneratedFiles" directory in the plugin work directory 
     // (which is unique for each plugin and target).
-    let outputDir = context.pluginWorkDirectory.appending("GeneratedFiles")
-    try FileManager.default.createDirectory(atPath: outputDir.string,
-          withIntermediateDirectories: true)
+    let outputDir = context.pluginWorkDirectoryURL
+        .appendingPathComponent("GeneratedFiles")
+    try FileManager.default.createDirectory(
+        at: outputDir,
+        withIntermediateDirectories: true)
 
     // Return a command to run `sometool` as a prebuild command. 
     // It runs before every build and generates source files 
@@ -217,26 +232,24 @@ In order to write a plugin that works with packages in every environment, and th
 For example:
 
 ```swift
-import PackagePlugin
-
-@main
-struct MyCommandPlugin: CommandPlugin {
-    /// This entry point is called when operating on a Swift package.
-    func performCommand(context: PluginContext,
-                        arguments: [String]) throws {
-        debugPrint(context)
-    }
-}
-
 #if canImport(XcodeProjectPlugin)
 import XcodeProjectPlugin
 
 extension MyCommandPlugin: XcodeCommandPlugin {
-    /// This entry point is called when operating on an Xcode project.
-    func performCommand(context: XcodePluginContext, 
-                        arguments: [String]) throws {
-        debugPrint(context)
+
+  // Entry point for creating build commands for targets in Xcode projects.
+  func createBuildCommands(context: XcodePluginContext,
+                           target: XcodeTarget) throws -> [Command] {
+    // Find the code generator tool to run (replace this with the actual one).
+    let generatorTool = try context.tool(named: "my-code-generator")
+
+    // Construct a build command for each source file with a particular suffix.
+    return target.inputFiles.map(\.url).compactMap {
+        createBuildCommand(for: $0,
+                           in: context.pluginWorkDirectoryURL,
+                           with: generatorTool.url)
     }
+  }
 }
 #endif
 ```

